tools/generator-go-sdk/generator/stage_predicates.go

```diff
@@ -20,7 +20,13 @@ func (s *ServiceGenerator) predicates(data ServiceGeneratorData) error {
 			continue
 		}
 
-		modelNames[*operation.ResponseObject.ReferenceName] = struct{}{}
+		modelName := *operation.ResponseObject.ReferenceName
+
+		if _, ok := data.models[modelName]; !ok {
```

Comment:
presumably we could output the predicates into the common types directory instead?